include/lsst/jointcal/FitterBase.h

```diff
@@ -0,0 +1,155 @@
+// -*- LSST-C++ -*-
+#ifndef LSST_JOINTCAL_FITTER_BASE_H
+#define LSST_JOINTCAL_FITTER_BASE_H
+
+#include "lsst/jointcal/Associations.h"
+#include "lsst/jointcal/CcdImage.h"
+#include "lsst/jointcal/Chi2.h"
+#include "lsst/jointcal/FittedStar.h"
+#include "lsst/jointcal/MeasuredStar.h"
+#include "lsst/jointcal/Tripletlist.h"
+
+namespace lsst {
+namespace jointcal {
+
+/**
+ * Base class for fitters.
+ *
+ * Implements minimize and findOutliers. Chi2, residual, derivative, etc. calculations must be
+ * implemented in the child class via the virtual methods.
+ */
+class FitterBase {
+public:
+    FitterBase(Associations &associations) : _associations(associations), _lastNTrip(0), _nMeasuredStars(0) {
+        // The various _npar are initialized in assignIndices, which must be called in the child constructor.
```

Comment:
How should one solve this? `FitterBase` doesn't know how to `assignIndices` at all, and I think I ran into trouble calling it from that constructor, instead of the child.